### PR TITLE
Configure Environment indicator.

### PR DESCRIPTION
**Pull request by sonnykt**
* Display an Orange "Local Development" message on local docker - .*.docker.amazee.io
* Display an Orange "Branch Development" message on all non-master branches - .*(?<!master).govcms.amazee.io
* Display a Green "Edit Domain" message on edit domains - (?<!www).govcms.gov.au
* Display a Red "Public Domain" message on public domains - .*.gov.au

### CODE REVIEW

File: .docker/images/govcms8/settings/production.settings.php [hunk 1]
```diff
@@ -12,14 +12,27 @@
...
+// even on production.
 $config['stage_file_proxy.settings']['origin'] = false;
+
+// Configure Environment indicator.
+$config['environment_indicator.indicator']['bg_color'] = 'red';
+$config['environment_indicator.indicator']['name'] = 'Public Domain';
+
+if (!empty($_SERVER['HTTP_HOST'])) {
+  $http_host = $_SERVER['HTTP_HOST'];
+  if (preg_match('/(?<!www)\.govcms\.gov\.au/i', $http_host)) {
```
Comment (fubarhouse) [inline .docker/images/govcms8/settings/production.settings.php]: Wondering if we can differentiate the actual govcms website from other customer websites?

For example:
`if (preg_match('/(?<!www)\.\.govcms\.gov\.au/i', $http_host)) {`

Would match anysite.govcms.gov.au and not govcms.gov.au, though given `www` is forced we might be able to dismiss the edge-case.

File: modules/lagoon/govcms_lagoon/govcms_lagoon.info.yml [hunk 2]
```diff
@@ -11,3 +11,4 @@ dependencies:
   - clamav:clamav
   - robotstxt:robotstxt
   - lagoon_logs:lagoon_logs
+  - environment_indicator:environment_indicator
```
Comment (simesy) [inline modules/lagoon/govcms_lagoon/govcms_lagoon.info.yml]: Does this have to be on by default?

**Comment by tobybellwood** [comment]
Recommendation from Drupal Slack
image

**Comment by srowlands** [comment]
Merging and revisiting colour selection in follow-up PR

**Comment by simesy** [comment]
@tobybellwood also from slack, let's accessiblilty check it.

Will need to be picked up for scaffold-tooling took

**Comment by brett-sargeant** [comment]
The current colours are not very readable, particularly in the sitename.govcms.gov.au environment  (orange background and #ddd for text), can they be set to [Toby's comment](https://github.com/govCMS/govcms8lagoon/pull/40#issuecomment-549195199)  instead.

image

**Comment by westerbail** [comment]
Hi Guys - I'm just following on from a Govcms support ticket that recommends tagging onto this request - we have been struggling with the colors from a readability standpoint and obviously there are options there to set up your own colors, but you can never export them/it only works locally - because it's locked down etc... I assume. 

We also have an issue where our 'master' URL is just not an acceptable URL for content editors to be using - so we have shortened govcms.gov.au editorial master URL set up for it (which shows the wrong colors) and also we don't allow editing on the gov.au domain because it will take an age for the akamai cache to update - so again no good for content editors.

Any suggestions etc.. would be great!

**Comment by simesy** [comment]
If this can't be overridden with config, and there was no accessibility review, I think this should be reverted.

**Comment by brett-sargeant** [comment]
Is there a text colour option that can be added around here? 
```
$config['environment_indicator.indicator']['bg_color'] = 'green';
$config['environment_indicator.indicator']['name'] = 'Branch Development';
```

If so, that is the easy solution as it is the text colour that is causing the problems.

**Comment by brett-sargeant** [comment]
> We also have an issue where our 'master' URL is just not an acceptable URL for content editors to be using - so we have shortened govcms.gov.au editorial master URL set up for it (which shows the wrong colors) and also we don't allow editing on the gov.au domain because it will take an age for the akamai cache to update - so again no good for content editors.


www.sitename.gov.au shouldn't be used for editing as the caching can do odd things other than just be slow. With the current colour settings, if you're in that environment it shows up as red. I assume that is to mean "unsafe, don't use".

The sitename.govcms.gov.au address which is where editing takes place is orange, so caution when making changes

Branches and local dev environment are green, so safe to make changes (within reason)

At least that is my assumption for the choice of colours.

**Comment by westerbail** [comment]
OK makes sense - test was showing as orange though, so I assumed red was a production and not a do not use..

**Comment by simesy** [comment]
> so I assumed red was a production and not a do not use..

I dislike the red for the same reason.

**Comment by stooit** [comment]
There is a follow-up PR here to simplify and bring AAA WCAG compliance: https://github.com/govCMS/govcms8lagoon/pull/154

**Comment by murraywoodman** [comment]
Thanks for fixing up the accessibility of the text. Big win!

**Pull request merged**